pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -166,6 +170,19 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, nil
 	}
 
+	// Check orgParentFolderID from configmap, create a new config map with the default value if it does not exist.
+	// If the valid configmap does not contain orgParentFolderID key, return err
+	if util.ConfigMapExist(r.client, operatorNamespace, orgGcpConfigMap) {
+		orgParentFolderID, err = util.GetGCPParentFolderFromConfigMap(r.client, operatorNamespace, orgGcpConfigMap)
+		if err != nil {
+			reqLogger.Error(err, "could not get orgParentFolderID from the ConfigMap:", orgGcpConfigMap, "Operator Namespace", operatorNamespace)
+			return reconcile.Result{}, err
+		}
+	} else {
+		configmap := util.NewConfigMapCR(orgGcpConfigMap, operatorNamespace, orgParentFolderID)
+		reqLogger.Info(fmt.Sprintf("%s has been created in Namespace: %s :: Moving with default orgParentFolderID", configmap.GetName(), operatorNamespace))
+	}
```

Comment:
As discussed, we can just return an error here, maybe even increase the time when this Project will be reconciled again, I think this is possible using reconcile.Result.RequeueAfter.  On a side note, please consider putting this into its own method.